src/main/java/htmlpublisher/HtmlPublisherTarget.java

```diff
@@ -179,18 +180,27 @@ protected File dir() {
         protected String getTitle() {
             return this.project.getDisplayName() + " html2";
         }
+        
+        /**
+         * Gets {@link HtmlPublisherTarget}, for which the action has been created.
+         * @return HTML Report description
+         * @since TODO
+         */
+        public @Nonnull HtmlPublisherTarget getHTMLTarget() {
+            return HtmlPublisherTarget.this;
+        }
     }
 
     public class HTMLBuildAction extends BaseHTMLAction {
-        private final AbstractBuild<?, ?> build;
+        private final Run<?, ?> build;
```

Comment:
In XML this thing appears to be <code>\<build class="build" reference="../../.."/\></code>, so it is not an issue. BTW, makes sense to change the behavior to prevent the entire build persistence on misuses.